cmd/clusterctl/client/repository/repository_github_test.go

```diff
@@ -207,28 +207,25 @@ func Test_gitHubRepository_getReleaseByTag(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			g, err := newGitHubRepository(providerConfig, configVariablesClient)
-			if err != nil {
-				t.Fatal(err)
-			}
-			g.injectClient = client
+			gRepo, err := newGitHubRepository(providerConfig, configVariablesClient)
+			g.Expect(err).NotTo(HaveOccurred())
 
-			got, err := g.getReleaseByTag(tt.args.tag)
-			if (err != nil) != tt.wantErr {
-				t.Fatalf("error = %v, wantErr %v", err, tt.wantErr)
-			}
+			gRepo.injectClient = client
+
+			got, err := gRepo.getReleaseByTag(tt.args.tag)
 			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
 				return
 			}
 
+			g.Expect(err).NotTo(HaveOccurred())
+
 			if tt.wantTagName == "" && got == nil {
 				return
 			}
 
 			if tt.wantTagName != "" && got != nil {
-				if *got.TagName != tt.wantTagName {
-					t.Errorf("got = %v, want %v", *got.TagName, tt.wantTagName)
-				}
+				g.Expect(*got.TagName).To(Equal(tt.wantTagName))
```

Comment:
I think this might clean up a bit of `wantTagName` was a *string rather than a string.

Reply:
made the change